modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RestRankEvalAction.java

```diff
@@ -164,22 +187,47 @@ public RestRankEvalAction(Settings settings, RestController controller, IndicesQ
     @Override
     public void handleRequest(final RestRequest request, final RestChannel channel, final NodeClient client) throws IOException {
         RankEvalRequest rankEvalRequest = new RankEvalRequest();
-        //parseRankEvalRequest(rankEvalRequest, request, parseFieldMatcher);
-        //client.rankEval(rankEvalRequest, new RestStatusToXContentListener<>(channel));
+        BytesReference restContent = RestActions.hasBodyContent(request) ? RestActions.getRestContent(request) : null;
+        try (XContentParser parser = XContentFactory.xContent(restContent).createParser(restContent)) {
+            QueryParseContext parseContext = new QueryParseContext(queryRegistry, parser, parseFieldMatcher);
+            if (restContent != null) {
+                parseRankEvalRequest(rankEvalRequest, request,
+                        new RankEvalContext(parseFieldMatcher, parseContext, aggregators, suggesters));
```

Comment:
Thanks for this comment - I had been wondering why we have those in here. I agree that this would be nice to simplify.